core_backend/app/voice_api/voice_components.py

```diff
@@ -12,13 +12,17 @@
 
 async def generate_speech(
     text: str,
-    language: IdentifiedLanguage,
+    language: IdentifiedLanguage | None,
     destination_blob_name: str = "response.mp3",
 ) -> str:
     """
     Converts the provided text to speech and saves it as an mp3 file on
     Google cloud storage
     """
+    if language is None:
+        error_msg = "Language must be provided to generate speech."
+        logger.error(error_msg)
+        raise ValueError(error_msg)
```

Comment:
I'd say keep the language parameter not-nullable in this function and move this check to the `_generate_tts_response` that calls it. This function should only be called if all of its parameters are available